packages/data-broker/src/__tests__/services/dhis/DhisService/testPull/testPullAnalytics.js

```diff
@@ -207,6 +207,15 @@ export const testPullAnalytics = () => {
         );
       });
 
+      it('non-existent program', async () => {
+        await expect(
+          dhisService.pull([DATA_SOURCES.POP01, DATA_SOURCES.POP02], 'dataElement', {
+            programCodes: ['POP01', 'DIFF_GROUP', 'CODE_NOT_IN_SYSTEM'],
+            useDeprecatedApi: false,
+          }),
+        ).rejects.toThrowError(/.*CODE_NOT_IN_SYSTEM.*/g);
```

Comment:
(No need to change this) you can also use our custom matcher:

```suggestion
        ).toBeRejectedWith(/.*CODE_NOT_IN_SYSTEM.*/g);
```

Those are defined in [/jest.setup.js](https://github.com/beyondessential/tupaia/blob/dev/jest.setup.js)